app/src/services/api.ts

```diff
@@ -261,6 +263,27 @@ export const api = createApi({
           response.data,
         providesTags: ["UserInfo"],
       }),
+      getUserPermissions: builder.query<
+        PermissionCheckResponse,
+        {
+          organizationId?: string;
+          projectId?: string;
+          cityId?: string;
+          inventoryId?: string;
+        }
+      >({
+        query: (params) => {
+          const searchParams = new URLSearchParams();
+          if (params.organizationId) searchParams.set('organizationId', params.organizationId);
+          if (params.projectId) searchParams.set('projectId', params.projectId);
+          if (params.cityId) searchParams.set('cityId', params.cityId);
+          if (params.inventoryId) searchParams.set('inventoryId', params.inventoryId);
```

Comment:
I think this could probably be expressed more succinctly by passing in an object and converting that to a URLSearchParams?